grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaChannel.java

```diff
@@ -142,7 +149,8 @@ public <I, O> ClientCall<I, O> newCall(
             }
         }
         if (credentials != null) {
-            client = new CallCredentialsDecoratingClient(httpClient, credentials, method, authority());
+            client = new CallCredentialsDecoratingClient(httpClient, credentials, method,
+                                                         callAuthority != null ? callAuthority : authority());
```

Comment:
nit; we don't check the emptiness of authority here which means the authority may be different. What do you think of just setting `callAuthority` to this value to start with?

```
final String callAuthority = Strings.isNullOrEmpty(callOptions.getAuthority()) ? authority() : callOptions.getAuthority() ;
```

Reply:
I wanted to leave `request.authority()` empty if `callAuthority` since the actual authority may get overridden by `DefautClientRequestContext.autoFillSchemeAndAuthority()`.
It is difficult to pass the actual authority to `CallCredentialsDecoratingClient` at this point. So I tried to pass possible an authority candidate to `CallCredentialsDecoratingClient`.

Reply:
OK then let's use `isNullOrEmpty` instead of just the nullable check. 😉